up-l1/mqtt.adoc

```diff
@@ -35,8 +35,128 @@ MQTT is an OASIS standard messaging protocol for the Internet of Things (IoT). I
 
 For more information, please refer to https://mqtt.org/
 
-
 == Specifications
 
+=== UAttribute Mapping
+
+MQTT 5 supports custom header fields, and we leverage this to map the UAttribute values into the MQTT header.
+
+[cols="1,1"]
+|===
+| MQTT Header Property Code | Value (key:value)
+| UserProperty
+| ("id", "{UAttribute.id}")
+| UserProperty
+| ("type", "{UAttribute.type}")
+| UserProperty
+| ("source","{UAttribute.source}") # Short Form Serialized
+| UserProperty
+| ("sink","{UAttribute.sink}") # Short Form Serialized
+| UserProperty
+| ("priority","{UAttribute.priority}")
+| CorrelationData
+| ("reqid", "{UAttribute.reqId}")
+| UserProperty
+| ("ttl", "{UAttribute.ttl}") # Optional
+| UserProperty
+| ("permission_level", "{UAttribute.permissionLevel}") # Optional
+| UserProperty
+| ("commstatus", "{UAttribute.commStatus}") # Optional
+| UserProperty
+| ("token", "{UAttribute.token}") # Optional
+| UserProperty
+| ("traceparent", "{UAttribute.traceParent}") # Optional
+|===
+
+
+Most uAttributes are mapped to a MQTT header UserProperty with the exception of the reqId which is mapped to the CorrelationData property.
+
+The source and sink fields **MUST** be a short form serialized UUri.
+
+The type field is used by a UListener to select message types on the topic.
+
+=== URI Mapping
+
+The URI mapping follows the short form UUri format, with the addition of a location identifier to determine if the message is intended for a local or cloud destination.
+
+
+- `up/{UUri.UAuthority.number}/{UUri.UEntity.id}/{UUri.UEntity.version_major}/{UUri.UResource.id}`
+
+  Note: If the UUri does not contain a UAuthority, add the UAuthority.number for the UAuthority the transport client is running on.
+
+Subscribe to all messages to the current Authority
+
+- `up/{UUri.UAuthority.number}/+/+/+` where `UAttributes.type != Publish`
```

Comment:
If we want this to be a URI then the scheme is `up:` and I would recommend omitting the scheme as I don't think we will have anything other than uProtocol topics right?

Reply:
Valid point, let me check that there are no concerns on starting off with a number

Reply:
removed